var/spack/repos/builtin/packages/py-scikit-learn/package.py

```diff
@@ -20,6 +20,11 @@ class PyScikitLearn(PythonPackage):
     version('0.17.1', 'a2f8b877e6d99b1ed737144f5a478dfc')
     version('0.13.1', 'acba398e1d46274b8470f40d0926e6a4')
 
-    depends_on('python@2.6:2.8,3.3:')
-    depends_on('py-numpy@1.6.1:', type=('build', 'run'))
-    depends_on('py-scipy@0.9:',   type=('build', 'run'))
+    depends_on('python@2.6:2.8,3.3:', when='@:0.19.1')
+    depends_on('python@2.7:2.8,3.4:', when='@0.20.0:')
+    depends_on('py-numpy@1.6.1:', type=('build', 'run'), when='@:0.19.1')
+    depends_on('py-numpy@1.8.2:', type=('build', 'run'), when='@0.20.0:')
+    depends_on('py-scipy@0.9:', type=('build', 'run'), when='@:0.19.1')
+    depends_on('py-scipy@0.13.3:', type=('build', 'run'), when='@0.20.0:')
+    depends_on('py-cython@0.23:')
```

Comment:
Probably `type='build'`.

Reply:
Would "import numpy" ... etc would work in that case when we load py-scikit-learn package? If not, then it might be a problem since, to my knowledge, py-scikit-learn is nearly "always" used together with numpy and scipy. Loading the exact same numpy,scipy used in compilation might be necessary when using py-scikit-learn.

Reply:
I'm only talking about `py-cython`, everything else should be `type=('build', 'run')`.